src/OrchardCore/OrchardCore.DisplayManagement/IShape.cs

```diff
@@ -105,10 +105,8 @@ public static TagBuilder GetTagBuilder(this IShape shape, string defaultTagName
 
             if (shape.Classes != null)
             {
-                foreach (var cssClass in shape.Classes)
-                {
-                    tagBuilder.AddCssClass(cssClass);
-                }
+                // Faster than AddCssClass which will do twice as many concatenations as classes.
+                tagBuilder.Attributes["class"] = string.Join(' ', shape.Classes);
```

Comment:
maybe event `shape.Classes != null && shape.Classes.Length > 0` so we don't call it at all if we don't have anything